### PR TITLE
feat: add GenAIEventsDictSessionMapper for dict spans with gen_ai events

### PR DESCRIPTION
**Pull request by stone-coding**
Description

  Adds GenAIEventsDictSessionMapper to handle dict spans with gen_ai semantic convention events (gen_ai.user.message, gen_ai.choice, gen_ai.system.message).

  The AgentCore evaluation service normalizes Strands telemetry spans into this format before passing them to evaluator Lambdas. Existing mappers cannot handle it:
  - CloudWatchSessionMapper requires body.input/output
  - StrandsInMemorySessionMapper requires ReadableSpan objects (not dicts)

  The new mapper:
  - Accepts list[dict] with gen_ai events
  - Extracts user_prompt, agent_response, system_prompt
  - Returns Session with AgentInvocationSpans
  - Is auto-detected by detect_otel_mapper() for Strands-scoped dict spans with gen_ai event names

  Known limitation: ToolExecutionSpan extraction is not yet supported for this format.

  Related Issues

  N/A — discovered during AgentCore SDK integration (bedrock-agentcore-sdk-python PR #568).

  Documentation PR

  N/A

  Type of Change

  New feature

  Testing

  - 13 unit tests covering detection logic, mapping, edge cases, and backward compatibility
  - Verified detect_otel_mapper() still returns CloudWatchSessionMapper for body format
  - Verified non-Strands scope spans are not incorrectly matched
  - [x] I ran hatch run prepare

  Checklist

  - [x] I have read the CONTRIBUTING document
  - [x] I have reviewed and understand every line of code in this PR, including any generated by AI tools, and I can explain why it works
  - [x] My change is focused and reasonably small; I have split unrelated work into separate PRs
  - [x] I have added any necessary tests that prove my fix is effective or my feature works
  - [ ] I have updated the documentation accordingly
  - [x] I have added an appropriate example to the documentation to outline the feature, or no new docs are needed
  - [x] My changes generate no new warnings
  - [x] Any dependent changes have been merged and published

  ---
  By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/strands_evals/mappers/gen_ai_events_dict_session_mapper.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+        return Session(traces=traces, session_id=session_id)
+
+    def _extract_agent_spans(
+        self, spans: list[dict[str, Any]], session_id: str, trace_id: str
+    ) -> list[SpanUnion]:
+        """Extract AgentInvocationSpans from dict spans with gen_ai events."""
+        agent_spans = []
+
+        for span in spans:
```
Comment (chianhw) [inline src/strands_evals/mappers/gen_ai_events_dict_session_mapper.py:80]: Can we check if the span is a agent span before extracting data?
for example:
```
span.attributes.get("gen_ai.operation.name") == "invoke_agent"
```

File: src/strands_evals/mappers/utils.py
Comment (chianhw) [inline src/strands_evals/mappers/utils.py:142]: PR 320 has been fixed this?
Can we pull down the latest mainline and apply the changes on top of it?

File: src/strands_evals/mappers/gen_ai_events_dict_session_mapper.py (hunk 1 above)
Comment (stone-coding) [reply src/strands_evals/mappers/gen_ai_events_dict_session_mapper.py:80]: Added attrs.get("gen_ai.operation.name") != "invoke_agent" check , only processes agent
  invocation spans, skips chat/tool spans.

File: src/strands_evals/mappers/utils.py
Comment (stone-coding) [reply src/strands_evals/mappers/utils.py:142]: Rebased on latest upstream/main (includes ADK mapper, latest fixes). PR #320 changes
  already there.

**Comment by JackYPCOnline** [comment]
Can you create a feature request to help better understand the use case? 
https://github.com/strands-agents/harness-sdk/blob/main/CONTRIBUTING.md
